pyvisa/ctwrapper/functions.py

```diff
@@ -832,6 +832,19 @@ def get_attribute(library, session, attribute):
         Return value of the library call.
 
     """
+    if attribute not in attributes.AttributesByID:
```

Comment:
This should not be needed and I would prefer to add the missing attribute definitions. I would also be curious to know how you found those and if this is something that could be integrated in the testsuite ?

Reply:
`from pyvisa.constants import ResourceAttribute
from pyvisa import VisaIOError
import pyvisa

rm = pyvisa.ResourceManager("@py")
scope = rm.open_resource("TCPIP::localhost::hislip0")

for attr in dir(ResourceAttribute):
    if attr.startswith("_"):
        continue
    try:
        value = scope.get_visa_attribute(getattr(ResourceAttribute, attr))
    except VisaIOError as e:
        value = str(e)
        if value.startswith("VI_ERROR_NSUP_ATTR"):
            continue
    except AttributeError as e:
        value = repr(e)
    except KeyError as e:
        value = repr(e)

    print(f"{attr}: {value!r}")
`
Maybe there could be a test case to verify that for all attributes in ResourceAttribute, get_visa_attribute() either returns a value or raises VisaIOError with an error message that starts with "VI_ERROR_NSUP_ATTR".

Reply:
We could add such a test but that requires a fake instrument and the dedicated worker is not in shape to run it and I do not have the bandwidth to fix it ATM.

Reply:
Fixed.